pluginframework/common/go_to_tf.go

```diff
@@ -0,0 +1,222 @@
+			continue
+		}
+
+		err := goSdkToTfSdkSingleField(srcField, destField, fieldInForceSendFields(srcFieldName, forceSendFieldVal), ctx)
+		if err != nil {
+			return diag.Diagnostics{diag.NewErrorDiagnostic(err.Error(), "gosdk to tfsdk field conversion failure")}
+		}
+	}
+	return nil
+}
+
+func goSdkToTfSdkSingleField(srcField reflect.Value, destField reflect.Value, forceSendField bool, ctx context.Context) error {
+
+	if !destField.CanSet() {
+		panic(fmt.Errorf("destination field can not be set: %s", destField.Type().Name()))
+	}
+
+	srcFieldValue := srcField.Interface()
+
+	if srcFieldValue == nil {
+		return nil
+	}
+
+	switch srcField.Kind() {
+	case reflect.Ptr:
+		if srcField.IsNil() {
+			// Skip nils
+			return nil
+		}
+		destField.Set(reflect.New(destField.Type().Elem()))
+
+		// Recursively populate the nested struct.
+		if diags := GoSdkToTfSdkStruct(srcFieldValue, destField.Interface(), ctx); diags.ErrorsCount() > 0 {
+			panic("Error converting gosdk to tfsdk struct")
+		}
+	case reflect.Bool:
+		boolVal := srcFieldValue.(bool)
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if boolVal || forceSendField {
+			destField.Set(reflect.ValueOf(types.BoolValue(boolVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.BoolNull()))
+		}
+	case reflect.Int64:
+		// convert any kind of integer to int64
+		intVal := srcField.Convert(reflect.TypeOf(int64(0))).Interface().(int64)
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if intVal != 0 || forceSendField {
+			destField.Set(reflect.ValueOf(types.Int64Value(int64(intVal))))
+		} else {
+			destField.Set(reflect.ValueOf(types.Int64Null()))
+		}
+	case reflect.Float64:
+		// convert any kind of float to float64
+		float64Val := srcField.Convert(reflect.TypeOf(float64(0))).Interface().(float64)
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if float64Val != 0 || forceSendField {
+			destField.Set(reflect.ValueOf(types.Float64Value(float64Val)))
+		} else {
+			destField.Set(reflect.ValueOf(types.Float64Null()))
+		}
+	case reflect.String:
+		var strVal string
+		if srcField.Type().Name() != "string" {
+			// This case is for Enum Types.
+			strVal = getStringFromEnum(srcField)
+		} else {
+			strVal = srcFieldValue.(string)
+		}
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if strVal != "" || forceSendField {
+			destField.Set(reflect.ValueOf(types.StringValue(strVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.StringNull()))
+		}
+	case reflect.Struct:
+		if srcField.IsZero() {
+			// Skip zeros
+			return nil
+		}
+		// resolve the nested struct by recursively calling the function
+		if GoSdkToTfSdkStruct(srcFieldValue, destField.Addr().Interface(), ctx).ErrorsCount() > 0 {
+			panic("Error converting gosdk to tfsdk struct")
+		}
+	case reflect.Slice:
+		if srcField.IsNil() {
```

Comment:
Here we don't need to worry about FSF because our API can't distinguish between no list and empty list in requests.